tools/system_libs.py

```diff
@@ -549,30 +549,59 @@ def get_default_variation(cls, **kwargs):
 
 class NoExceptLibrary(Library):
   def __init__(self, **kwargs):
-    self.is_noexcept = kwargs.pop('is_noexcept')
+    self.eh_mode = kwargs.pop('eh_mode')
+    if self.eh_mode not in ('noexcept', 'emscripten', 'wasm'):
+      raise Exception('Invalid exception mode')
```

Comment:
this might be an assertion (which is shorter, on a single line)? but might not be needed if we can avoid adding `eh_mode` as mentioned in the other comment.

Reply:
I'd like to keep `eh_mode` (reason explained in the other comment), so I changed this to an assertion. If we end up deleting `eh_mode`, I'll delete this then.